Client/AccessoryViewProvider.swift

```diff
@@ -0,0 +1,144 @@
+                                                      style: .plain,
+                                                      target: target,
+                                                      action: nil)
+
+    lazy private var nextButton = UIBarButtonItem(image: UIImage(systemName: "chevron.down"),
+                                                  style: .plain,
+                                                  target: target,
+                                                  action: nil)
+
+    lazy private var doneButton = UIBarButtonItem(title: "Done",
+                                                  style: .done,
+                                                  target: target,
+                                                  action: nil)
+
+    private let flexibleSpacer = UIBarButtonItem(systemItem: .flexibleSpace)
+
+    private let fixedSpacer: UIView = .build { view in
+        NSLayoutConstraint.activate([
+            view.widthAnchor.constraint(equalToConstant: 2),
+            view.heightAnchor.constraint(equalToConstant: 30)
+        ])
+
+        view.accessibilityElementsHidden = true
+    }
+
+    lazy private var cardImageView: UIImageView = .build { imageView in
+        imageView.image = UIImage(named: ImageIdentifiers.creditCardPlaceholder)?.withRenderingMode(.alwaysTemplate)
+        imageView.contentMode = .scaleAspectFit
+        imageView.accessibilityElementsHidden = true
+
+        NSLayoutConstraint.activate([
+            imageView.widthAnchor.constraint(equalToConstant: 24),
+            imageView.heightAnchor.constraint(equalToConstant: 24)
+        ])
+    }
+
+    lazy private var useCardTextLabel: UILabel = .build { label in
+        label.font = DynamicFontHelper.defaultHelper.preferredFont(withTextStyle: .title3, size: 16, weight: .medium)
+        label.text = .CreditCard.Settings.UseSavedCardFromKeyboard
+        label.numberOfLines = 0
+    }
+
+    private lazy var cardButtonStackView: UIStackView = .build { [weak self] stackView in
+        guard let self = self else { return }
+
+        stackView.addArrangedSubview(self.fixedSpacer)
+        stackView.addArrangedSubview(self.cardImageView)
+        stackView.addArrangedSubview(self.useCardTextLabel)
+        stackView.addArrangedSubview(self.fixedSpacer)
+        stackView.spacing = 2
+        stackView.distribution = .equalCentering
+    }
+
+    init(for target: TabWebView,
+         themeManager: ThemeManager = AppContainer.shared.resolve(),
+         notificationCenter: NotificationCenter = NotificationCenter.default) {
+        self.target = target
+        self.themeManager = themeManager
+        self.notificationCenter = notificationCenter
+
+        super.init(frame: CGRect(width: UIScreen.main.bounds.width,
+                                 height: AccessoryViewUX.toolbarHeight))
+
+        listenForThemeChange(self)
+        setupLayout()
+        applyTheme()
+
+        self.notificationCenter.addObserver(self,
```

Comment:
The only time that notification is posted is when the script executes and `foundFieldValues` closure is run, which is currently only when a user taps on a text field that is a CC input field. I'm not sure how the view would get overwhelmed with notifications in this case 🤔 . I'm open to trying the suggested approach as well, but not understanding the reason we need to avoid a notification based approach yet.

Reply:
I don't want to rely on Notifications for such update where we can do a better job at handling how we update a view as its 1~1 update rather than 1~many